src/parquet/util/input.h

```diff
@@ -183,6 +184,25 @@ class InMemoryInputStream : public InputStream {
   int64_t offset_;
 };
 
+// Implementation of an InputStream when only some of the bytes are in memory.
+class ChunkedInMemoryInputStream : public InputStream {
```

Comment:
Can we call this a `BufferedInputStream`? Also, change the name of `chunk_size` and related arguments to `buffer_size`. We should document that while the passed `buffer_size` is the minimum size of the buffer, it will be grown (and, at the moment, never shrunk) as required based on the sizes of reads requested by the user.